src/scan-dirs.ts

```diff
@@ -11,21 +11,27 @@ export async function scanDirExports (dir: string | string[], options?: ScanDirE
   const fileFilter = options?.fileFilter || (() => true)
   const filePatterns = options?.filePatterns || ['*.{ts,js,mjs,cjs,mts,cts}']
 
-  const scanDirs = dirs.map((i) => {
-    return fg([i].concat(filePatterns.map(p => join(i, p))), {
-      absolute: true,
-      cwd: options?.cwd || process.cwd(),
-      onlyFiles: true,
-      followSymbolicLinks: true
-    })
-  })
-
-  const result = await Promise.all(scanDirs)
-  const files = result.flatMap(f => f.map(f => normalize(f)).sort()).filter(fileFilter)
-  const fileExports = await Promise.all(
-    files.map(scanExports)
+  const result = await Promise.all(
+    // Do multiple glob searches to persist the order of input dirs
+    dirs.map(async i => await fg(
+      [i, ...filePatterns].map(p => join(i, p)),
```

Comment:
there's a mistake here, it should be `[i, ...filePatterns.map(p => join(i, p))]` instead

Reply:
Thanks! Fixed in v1.0.1